modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/LayoutReportsApp.js

```diff
@@ -12,33 +12,166 @@
  * details.
  */
 
-import React from 'react';
+import ClayAlert from '@clayui/alert';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import {useIsMounted} from '@liferay/frontend-js-react-web';
+import {fetch} from 'frontend-js-web';
+import React, {
+	useCallback,
+	useEffect,
+	useLayoutEffect,
+	useReducer,
+} from 'react';
 
 import BasicInformation from './components/BasicInformation';
 import EmptyLayoutReports from './components/EmptyLayoutReports';
 
-export default function ({context}) {
-	const {
-		assetsPath,
-		canonicalURLs,
-		configureGooglePageSpeedURL,
-		defaultLanguageId,
-		validConnection,
-	} = context;
-
-	return (
-		<>
-			<BasicInformation
-				canonicalURLs={canonicalURLs}
-				defaultLanguageId={defaultLanguageId}
-			/>
-
-			{validConnection || (
-				<EmptyLayoutReports
-					assetsPath={assetsPath}
-					configureGooglePageSpeedURL={configureGooglePageSpeedURL}
+const initialState = {
+	data: null,
+	error: null,
+	loading: false,
+};
+
+const dataReducer = (state, action) => {
+	switch (action.type) {
+		case 'LOAD_DATA':
+			return {
+				...state,
+				loading: true,
+			};
+
+		case 'SET_ERROR':
+			return {
+				...state,
+				error: action.error,
+				loading: false,
+			};
+
+		case 'SET_DATA':
+			return {
+				data: {
+					...action.data,
+				},
+				error: action.data?.error,
+				loading: false,
+			};
+
+		default:
+			return initialState;
+	}
+};
+
+export default function ({
+	isPanelStateOpen,
+	layoutReportsDataURL,
+	portletNamespace,
+}) {
+	const isMounted = useIsMounted();
+
+	const [state, dispatch] = useReducer(dataReducer, initialState);
+
+	const safeDispatch = (action) => {
+		if (isMounted()) {
+			dispatch(action);
+		}
+	};
+
+	const getData = (fetchURL) => {
+		safeDispatch({type: 'LOAD_DATA'});
+
+		fetch(fetchURL, {
+			method: 'GET',
+		})
+			.then((response) =>
+				response.json().then((data) =>
+					safeDispatch({
+						data,
+						type: 'SET_DATA',
+					})
+				)
+			)
+			.catch(() => {
+				safeDispatch({
+					error: Liferay.Language.get('an-unexpected-error-occurred'),
+					type: 'SET_ERROR',
+				});
+			});
+	};
+
+	const handlePageAuditData = useCallback(() => {
+		getData(layoutReportsDataURL);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Instead of suppressing the lint (bad habit, because even if you know it's safe here, it becomes a pattern that people unthinkingly apply in other places where it's not safe), I'd say use `useCallback` on the upstream functions `getData` (and `safeDispatch`), and add `getData` to the dependencies array.

Reply:
Ok, I'll try